### DISCUSSION TITLE
How can I contribute to the Avalonia rewrite?

### DISCUSSION THREAD

**Discussion by martinrhan** [stride3d/stride#2160] [Contributors]
I am really looking forward for the Avalonia rewrite as I really want a fully featured add-on or plugin system. However, I don't see much progress on that. Only one person is working on that @Kryptos-FR 
I would like to know if someone else want to contribute, how can the additional contributer help? How are tasks going to allocated?

**Comment by tebjan** [comment]
Hey there!

Great to hear you're interested in helping with the Avalonia rewrite! We're all looking forward to seeing how it evolves and the new possibilities it opens up, especially for a fully featured add-on or plugin system.

You're right that @Kryptos-FR has been steering this ship solo for a bit. Any extra hands are definitely welcome, and your interest in contributing is super appreciated!

The best way to jump in would be to [join our Discord server](https://discord.gg/f6aerfE). Once you're there, give @Kryptos-FR a ping. He'll be able to fill you in on where things stand and discuss how you can best contribute. There's always a mix of tasks to tackle, from simple to complex, so there's a good chance we can find something that matches your skills and interests.

Looking forward to seeing you on Discord and working together to push the Avalonia rewrite forward!

Cheers!

**Comment by VaclavElias** [comment]
I can count at least 3-5 people who would like to contribute, at least, they expressed an interest on Discord. 

Kryptos is preparing foundation, so others can join, his last comments are also here https://discord.com/channels/500285081265635328/1163440177788768377

**Comment by Kryptos-FR** [comment]
Glad to see some interest in it. I have some refactoring to do (which is not easily split into tasks that several people can work on unfortunately). Only then can we consider having features worked on in parallel. I'll have more time next week to work on it.
